backend/app.py

```diff
@@ -204,26 +220,16 @@ def create_machine():
         float(request.args.get("lat_coord")),
     )
     # Verify that address matches coordinates
-    queries = [address, address + area, address + title]
-    for query in queries:
-        coordinates = GM_CLIENT.geocode(query)
-        try:
-            lat = coordinates[0]["geometry"]["location"]["lat"]
-            lng = coordinates[0]["geometry"]["location"]["lng"]
-            break
-        except IndexError:
-            continue
-    try:
-        lat, lng
-    except NameError:
+    found_coords, (lat, lng) = verify_machine_location(address, area, title)
+    if not found_coords:
         return jsonify({"error": "Google Maps does not know this address"}), 400
 
     dist = haversine((lat, lng), (location[1], location[0]))
     if dist > 1:  # km
         return (
             jsonify(
                 {
-                    "error": f"Address {query} seems >1km away from coordinates ({lat}, {lng})"
+                    "error": f"Address {address} seems >1km away from coordinates ({location[1]}, {location[0]})"
```

Comment:
I think this was an error previously. The user should see the original coordinates in the error message, not the ones that were found after geocoding the address.

Reply:
Good catch!